### PR TITLE
React to HttpResponse.Challenge API changes

### PR DESCRIPTION
**Pull request by HaoK**
https://github.com/aspnet/HttpAbstractions/pull/254

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Core.Test/ActionResults/ChallengeResultTest.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public void ChallengeResult_Execute()
             result.ExecuteResult(actionContext);
 
             // Assert
-            httpResponse.Verify(c => c.Challenge(null, (IEnumerable<string>)new string[] { }), Times.Exactly(1));
+            httpResponse.Verify(c => c.Challenge(null, ""), Times.Exactly(1));
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/ActionResults/ChallengeResultTest.cs]: We should have a new test to cover the other behavior (no schemes)?

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/ChallengeResult.cs [hunk 2]
```diff
@@ -47,7 +48,17 @@ public ChallengeResult(IList<string> authenticationSchemes, AuthenticationProper
         public override void ExecuteResult([NotNull] ActionContext context)
         {
             var response = context.HttpContext.Response;
-            response.Challenge(Properties, AuthenticationSchemes);
+            if (AuthenticationSchemes.Any())
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/ChallengeResult.cs]: Count > 0

**Comment by HaoK** [comment]
Updated with PR feedback

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/ChallengeResult.cs [hunk 3]
```diff
@@ -47,7 +47,17 @@ public ChallengeResult(IList<string> authenticationSchemes, AuthenticationProper
         public override void ExecuteResult([NotNull] ActionContext context)
         {
             var response = context.HttpContext.Response;
-            response.Challenge(Properties, AuthenticationSchemes);
+            if (AuthenticationSchemes.Count > 0)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/ChallengeResult.cs]: Will this every be null? Might want a `?.` in there #PranavPoints

**Comment by rynowak** [comment]
:shipit:

**Comment by HaoK** [comment]
bee20973c770c0bf6af27f8aa50c45206e89111e

**Pull request closed without merging**